csrc/cpu_ops.cpp

```diff
@@ -23,15 +27,19 @@ void quantize_cpu(float *code, float *A, float *absmax, unsigned char *out, long
     num_blocks += n % blocksize == 0 ? 0 : 1;
 
     const uint32 elements_code = 256;
-    BinAlgo<Scalar, float, Direct2> bin_searcher(code, elements_code);
+    BinAlgo<INSTR_SET, float, Direct2> bin_searcher(code, elements_code);
 
     int thread_wave_size = 256;
     // we chunk the thresds into waves of 256 since the max limit is
     // between 16k and 64k on Linux (we reach this when running BLOOM-176B with a large batch size)
     for(long long offset = 0; offset < num_blocks; offset+=thread_wave_size)
     {
       long long valid_chunks = num_blocks - offset >= thread_wave_size ? thread_wave_size : num_blocks - offset;
+#ifdef _WIN32
+      std::thread *threads = (std::thread *) malloc(sizeof(std::thread) * valid_chunks);
```

Comment:
Any reason not to use std::thread everywhere?

Reply:
> Any reason not to use std::thread everywhere?

I just leaving this as the original patch creator intended.